pkg/gitops_test.go

```diff
@@ -36,17 +36,11 @@ func TestCloneGenerateAndPush(t *testing.T) {
 	repoPath := "/fake/path/test-component"
 	componentName := "test-component"
 	component := gitopsv1alpha1.Component{
-		Spec: gitopsv1alpha1.ComponentSpec{
-			ContainerImage: "testimage:latest",
-			Source: gitopsv1alpha1.ComponentSource{
-				ComponentSourceUnion: gitopsv1alpha1.ComponentSourceUnion{
-					GitSource: &gitopsv1alpha1.GitSource{
-						URL: repo,
-					},
-				},
-			},
-			TargetPort: 5000,
+		ContainerImage: "testimage:latest",
+		Source: &gitopsv1alpha1.ComponentSource{
+			URL: repo,
```

Comment:
Add coverage for ComponentSource optional fields

Reply:
Will remove those fields.